### PR TITLE
Add E2E tests for OTA suspend and resume

### PR DESCRIPTION
**Pull request by tgsong**
Description
-----------

Add 2 E2E tests for OTA suspend and resume feature.

1. Disconnects the device from IoT core once. Device should reconnect and resume OTA update.
1. Disconnects the device from IoT core once and cancel the OTA job. Device should reconnect, abort the update and then go back to normal state. Do another normal OTA update, device should be able to update successfully.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tools/ota_e2e_tests/aws_ota_test/aws_ota_test_case_disconnect_cancel_update.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import time
+from pathlib import Path
+import boto3
+from botocore.exceptions import ClientError
+import paho.mqtt.client as mqtt
+from .aws_ota_test_case import OtaTestCase
+from .aws_ota_test_result import OtaTestResult
+
+
+class OtaTestDisconnectCancelUpdate(OtaTestCase):
```
Comment (dcgaws) [inline tools/ota_e2e_tests/aws_ota_test/aws_ota_test_case_disconnect_cancel_update.py:35]: Could there be a block comment here that summarizes what this test case does, and how and why?

File: tools/ota_e2e_tests/aws_ota_test/aws_ota_test_case_disconnect_resume.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import time
+from pathlib import Path
+import boto3
+from botocore.exceptions import ClientError
+import paho.mqtt.client as mqtt
+from .aws_ota_test_case import OtaTestCase
+from .aws_ota_test_result import OtaTestResult
+
+
+class OtaTestDisconnectResume(OtaTestCase):
```
Comment (dcgaws) [inline tools/ota_e2e_tests/aws_ota_test/aws_ota_test_case_disconnect_resume.py:35]: Ditto?

File: tools/ota_e2e_tests/aws_ota_test/aws_ota_test_case_back_to_back_downloads.py [hunk 3]
```diff
@@ -28,6 +28,10 @@
 
 
 class OtaTestBackToBackDownloads(OtaTestCase):
+    """
+    This test creats 3 consecutive OTA updates. The device is expected to update 3 times in a row.
```
Comment (pvyawaha) [inline tools/ota_e2e_tests/aws_ota_test/aws_ota_test_case_back_to_back_downloads.py:32]: Spelling creates?

File: tools/ota_e2e_tests/aws_ota_test/aws_ota_test_case_unsigned_image.py [hunk 4]
```diff
@@ -28,6 +28,10 @@
 
 
 class OtaTestUnsignedImage(OtaTestCase):
+    """
+    This test verifies that device will reject an update if the image signed.
```
Comment (pvyawaha) [inline tools/ota_e2e_tests/aws_ota_test/aws_ota_test_case_unsigned_image.py:32]: It should be image is unsigned or not signed?

**Pull request merged**